### ISSUE TITLE
fs.move deletes original file and throws error failing to rename/move

### ISSUE THREAD

**Issue by kdcro101** [jprichardson/node-fs-extra#614]
- **Operating System: Windows 10**
- **Node.js version: 10.2.1**
- **fs-extra version: 6.0.0, 7.0.0**

when renaming (using `fs.move + overwrite:true`) file from: `C:\DIR\file.txt` to `C:\DIR\FILE.txt` (from lowercase to uppercase and vice versa) original file is deleted and error is thrown that file doesn't exist.

On linux everything works as expected.

**Comment by manidlou** [comment]
ooh yeah I know what's happening here! @RyanZim @kdcro101 Windows is a case-insensitive system meaning in Windows world `file.txt` and `FILE.txt` are the same and since `{overwrite: true}` the dest file gets removed while src and dest are actually the same, therefore we end up removing src file!

We applied the fix for handling case-insensitive paths to `copy*()` but `move*()` is still buggy and needs to be fixed. I'll be working on it. I do my best to make the fix ready for v8 release.

**Comment by bluelovers** [comment]
a stupid way

```ts
import * as fs from 'fs-extra';

export function win32_move(src: string, dest: string, options?: fs.MoveOptions)
{
	return fs.move(src, dest, options)
		.catch(function (e)
		{
			if (process.platform ==='win32')
			{
				if (/dest already exists/i.test(e.message))
				{
					if (src.toLowerCase() === dest.toLowerCase())
					{
						return;
					}
				}
			}

			return Promise.reject(e)
		})
	;
}
```

**Comment by RyanZim** [comment]
Should be fixed in #667
